vllm/transformers_utils/tokenizer.py

```diff
@@ -145,6 +216,94 @@ async def get_lora_tokenizer_async(
             return self.lora_tokenizers.get(lora_request.lora_int_id)
 
 
+if ray:
+    RayTokenizerGroup = ray.remote(TokenizerGroup)
+
+    class RayTokenizerGroupPool(BaseTokenizerGroup):
+        """A pool of TokenizerGroups for async tokenization."""
```

Comment:
Why do we have a pool of groups? (probably I'm missing some detail)

Reply:
I'm confused between the two -- we use `RayTokenizerGroupPool` vs just `TokenizerGroup` in the engine

Reply:
One group contains multiple HF tokenizers, but is fully sequential (it's basically an abstraction over a dict of tokenizers so the right one can be chosen for a given LoRA). A pool contains multiple workers which hold a group